spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/ComposedRunnerJobFactory.java

```diff
@@ -352,4 +356,13 @@ private Flow getTaskAppFlow(TaskAppNode taskApp) {
 
 		return new FlowBuilder<Flow>(beanName).from(currentStep).end();
 	}
+
+	public static class UuidIncrementer implements JobParametersIncrementer {
+
+		@Override
+		public JobParameters getNext(JobParameters parameters) {
+			JobParameters params = (parameters == null) ? new JobParameters() : parameters;
+			return new JobParametersBuilder(params).addString(CTR_KEY, UUID.randomUUID().toString()).toJobParameters();
```

Comment:
So this doesn't cause any trouble that key is suddenly changed from `run.id` to `ctr.id` and type from long to string?

Reply:
Not for batch or CTR.   However, if users are somehow useing the run.id property it could be an issue for them.   We could add a property that allows users to enable the use of UUID versus the default run id incrementer.   Thoughts?